src/server/journal/serializer.cc

```diff
@@ -18,154 +18,182 @@ using namespace std;
 
 namespace dfly {
 
-JournalWriter::JournalWriter(io::Sink* sink, std::optional<DbIndex> dbid)
-    : sink_{sink}, cur_dbid_{dbid} {
+JournalWriter::JournalWriter(std::optional<DbIndex> dbid) : cur_dbid_{dbid} {
 }
 
-error_code JournalWriter::Write(uint64_t v) {
+std::error_code JournalWriter::Flush(io::Sink* sink) {
+  if (auto ec = sink->Write(buf_.InputBuffer()); ec)
+    return ec;
+  buf_.Clear();
+  return {};
+}
+
+io::Bytes JournalWriter::Accumulated() {
+  return buf_.InputBuffer();
+}
+
+void JournalWriter::Write(uint64_t v) {
   uint8_t buf[10];
   unsigned len = WritePackedUInt(v, buf);
-  return sink_->Write(io::Bytes{buf, len});
+  buf_.EnsureCapacity(10);
```

Comment:
it will be better if this magic 10 number will be defined next to WritePackedUInt

Reply:
or use `sizeof(buf)`